server/src/tables.rs

```diff
@@ -40,240 +47,361 @@ use datafusion::physical_plan::{
-                    if **left == col("tid") {
-                        // Assumes time series are not grouped so tids and gids are equivalent.
-                        self.binary_expr(col("gid"), *op, *right.clone())
-                    } else if **left == col("timestamp") {
-                        match op {
-                            Operator::Gt => {
-                                self.binary_expr(col("end_time"), *op, self.to_i64(right))
-                            }
-                            Operator::GtEq => {
-                                self.binary_expr(col("end_time"), *op, self.to_i64(right))
-                            }
-                            Operator::Lt => {
-                                self.binary_expr(col("start_time"), *op, self.to_i64(right))
-                            }
-                            Operator::LtEq => {
-                                self.binary_expr(col("start_time"), *op, self.to_i64(right))
-                            }
-                            Operator::Eq => self.binary_expr(
-                                self.binary_expr(
-                                    col("start_time"),
-                                    Operator::LtEq,
-                                    self.to_i64(right),
-                                ),
-                                Operator::And,
-                                self.binary_expr(
-                                    col("end_time"),
-                                    Operator::GtEq,
-                                    self.to_i64(right),
-                                ),
-                            ),
-                            _ => filter.clone(),
-                        }
-                    } else {
-                        filter.clone()
-                    }
-                }
-                _ => filter.clone(),
-            })
-            .collect();
-        combine_filters(&rewritten_filters)
+    // TODO: Move to the metadata component when it exists.
+    /// Compute the 64-bit keys of the univariate time series to retrieve from
+    /// the storage engine using the fields, tag, and tag values in the query.
+    /// Returns a [`DataFusionError::Plan`] if the necessary data cannot be
+    /// retrieved from [`catalog::METADATA_SQLITE_NAME`].
+    fn compute_keys_using_fields_and_tags(
+        &self,
+        table_name: &str,
+        columns: &Option<Vec<usize>>,
+        tag_predicates: &[(&str, &str)],
+    ) -> Result<Vec<TimeSeriesId>> {
+        // Compute the location of the database containing the metadata.
+        let database_path = {
+            // unwrap() is safe as read() only fails if the RwLock is poisoned.
+            let catalog = self.context.catalog.read().unwrap();
+            catalog.data_folder_path.join(catalog::METADATA_SQLITE_NAME)
+        };
+
+        // Construct the two queries that extract the field columns in the table
```

Comment:
Very minor but maybe move half of this comment down to the construction of the second query so the explanation of what it does is right above the query.l

Reply:
Fixed.